### ISSUE TITLE
False positive `unused-private-member` if class member is accessed via `self`

### ISSUE THREAD

**Issue by akosthekiss** [PyCQA/pylint#4644]
### Steps to reproduce

Given a file `p.py`:
```python
"""p.py"""

class Klass:  # pylint: disable=too-few-public-methods
    """Klass"""

    __seventyseven = 77
    __ninetyone = 91

    def __init__(self):
        self.twentyone = (self.__seventyseven + 44) % 100
        self.ninetyfive = Klass.__ninetyone + 4

k = Klass()
print(k.twentyone)
print(k.ninetyfive)
```

### Current behavior

Result of `pylint p.py`:
```
************* Module p
p.py:6:4: W0238: Unused private member `Klass.__seventyseven` (unused-private-member)
```

### Expected behavior

No diagnostics.

### pylint --version output

Result of `pylint --version` output:

```
pylint 2.9.1
astroid 2.6.2
Python 3.6.9 (default, Jan 26 2021, 15:33:00) 
[GCC 8.4.0]
```

```
pylint 3.0.0-a4
astroid 2.6.2
Python 3.6.9 (default, Jan 26 2021, 15:33:00) 
```

**Comment by Pierre-Sassoulas** [comment]
Thank you, I can reproduce this.

**Comment by Pierre-Sassoulas** [comment]
This will be fixed in ``2.9.2``, we'll release it when https://github.com/PyCQA/pylint/issues/4630 is merged :)

**Comment by earonesty** [comment]
I get this error on classmethods no matter what, regardless of self-use and no mutation needed:

Snippet:

```
    @classmethod
    def __get_type(cls, k) -> Type:						# pylint: disable=unused-private-member
        return cls.__annotations__.get(k, None)

    @staticmethod
    def __accept_instance(v, typ):                      # pylint: disable=unused-private-member
        if typ is Any:
            return True
        if type(v) is int and issubclass(typ, float):
            return True
        try:
            return isinstance(v, typ)
        except TypeError:
            # not checking types i don't know how to handle
            return True

    @classmethod
    def __assert_instance(cls, k, v):
        typ = cls.__get_type(k)
        if getattr(typ, "__origin__", None) is Union:
            for sub in typ.__args__:
                if cls.__accept_instance(v, sub):
                    return
```

**Comment by vepain** [comment]
Dear contributors,

this issue is still here in `Pylint 3.0.0`

**Versions:**

```bash
$ pylint --version
pylint 3.0.0-a4
astroid 2.6.6
Python 3.9.5 (default, May 19 2021, 11:32:47) 
[GCC 9.3.0]
```

**MWE:**

```python
class A():

    __CLS_ATTR = 0  # pylint: unused-private-member

    def __init__(self, t_int_int: tuple[int, int]):
        self._object_attr: tuple[int, int] = t_int_int
     
    def get_first_int(self) -> int:
        return self._object_attr[self.__CLS_ATTR]
```

Thank your for your help !

**Comment by Pierre-Sassoulas** [comment]
Hey, it's because 3.0 alpha is pylint 2.8 with some breaking change rebased on it. I might release a new 3.0 alpha rebased on 2.11 when we fix the problem with the current alpha version.
